<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Sass Errors - v0.4.0</title><link rel="stylesheet" href="assets/css/main.css"><link href="https://fonts.googleapis.com/css?family=Open+Sans:400,500,700" rel="stylesheet" type="text/css"><meta name="viewport" content="width=device-width"><meta content="IE=edge, chrome=1" http-equiv="X-UA-Compatible"><!-- Open Graph tags --><meta property="og:title" content="Sass Errors - SassDoc"><meta property="og:type" content="website"><meta property="og:description" content="<p>âœ¨ Awesome Error Handling in Sass</p>
"><!-- Thanks to Sass-lang.com for the icons --><link href="assets/images/favicon.png" rel="shortcut icon"></head><body><aside class="sidebar" role="nav"><div class="sidebar__header"><h1 class="sidebar__title"><a href="https://github.com/ncoden/sass-errors#readme">Sass Errors - v0.4.0</a></h1></div><div class="sidebar__body"><button type="button" class="btn-toggle js-btn-toggle" data-alt="Open all">Close all</button><p class="sidebar__item sidebar__item--heading" data-slug="alias"><a href="#alias">Aliases</a></p><div><p class="sidebar__item sidebar__item--sub-heading" data-slug="alias-function"><a href="#alias-function">functions</a></p><ul class="list-unstyled"><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-init" data-type="function"><a href="#alias-function-se-init">se-init <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-store" data-type="function"><a href="#alias-function-se-store">se-store <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-message" data-type="function"><a href="#alias-function-se-message">se-message <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-new" data-type="function"><a href="#alias-function-se-new">se-new <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-throw" data-type="function"><a href="#alias-function-se-throw">se-throw <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-try" data-type="function"><a href="#alias-function-se-try">se-try <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-catch" data-type="function"><a href="#alias-function-se-catch">se-catch <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-try-catch" data-type="function"><a href="#alias-function-se-try-catch">se-try-catch <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-get" data-type="function"><a href="#alias-function-se-get">se-get <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-is-exception" data-type="function"><a href="#alias-function-se-is-exception">se-is-exception <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-debug" data-type="function"><a href="#alias-function-se-debug">se-debug <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-error" data-type="function"><a href="#alias-function-se-error">se-error <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li><li class="sidebar__item sassdoc__item" data-group="alias" data-name="se-warn" data-type="function"><a href="#alias-function-se-warn">se-warn <span class="sidebar__annotation sidebar__annotation--alias">alias</span></a></li></ul></div><p class="sidebar__item sidebar__item--heading" data-slug="api"><a href="#api">API</a></p><div><p class="sidebar__item sidebar__item--sub-heading" data-slug="api-function"><a href="#api-function">functions</a></p><ul class="list-unstyled"><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-catch" data-type="function"><a href="#api-function-sass-errors-catch">sass-errors-catch</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-get" data-type="function"><a href="#api-function-sass-errors-get">sass-errors-get</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-init" data-type="function"><a href="#api-function-sass-errors-init">sass-errors-init</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-is-exception" data-type="function"><a href="#api-function-sass-errors-is-exception">sass-errors-is-exception</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-message" data-type="function"><a href="#api-function-sass-errors-message">sass-errors-message</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-new" data-type="function"><a href="#api-function-sass-errors-new">sass-errors-new</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-store" data-type="function"><a href="#api-function-sass-errors-store">sass-errors-store</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-throw" data-type="function"><a href="#api-function-sass-errors-throw">sass-errors-throw</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-try-catch" data-type="function"><a href="#api-function-sass-errors-try-catch">sass-errors-try-catch</a></li><li class="sidebar__item sassdoc__item" data-group="api" data-name="sass-errors-try" data-type="function"><a href="#api-function-sass-errors-try">sass-errors-try</a></li></ul></div><p class="sidebar__item sidebar__item--heading" data-slug="handler"><a href="#handler">Exception Handlers</a></p><div><p class="sidebar__item sidebar__item--sub-heading" data-slug="handler-function"><a href="#handler-function">functions</a></p><ul class="list-unstyled"><li class="sidebar__item sassdoc__item" data-group="handler" data-name="sass-errors-debug" data-type="function"><a href="#handler-function-sass-errors-debug">sass-errors-debug</a></li><li class="sidebar__item sassdoc__item" data-group="handler" data-name="sass-errors-error" data-type="function"><a href="#handler-function-sass-errors-error">sass-errors-error</a></li><li class="sidebar__item sassdoc__item" data-group="handler" data-name="sass-errors-warn" data-type="function"><a href="#handler-function-sass-errors-warn">sass-errors-warn</a></li></ul></div><p class="sidebar__item sidebar__item--heading" data-slug="undefined"><a href="#undefined">General</a></p><div><p class="sidebar__item sidebar__item--sub-heading" data-slug="undefined-css"><a href="#undefined-css">csss</a></p><ul class="list-unstyled"><li class="sidebar__item sassdoc__item" data-group="undefined" data-name="/// If an Exception has been thrown, catch it with `sass-errors-catch` and pass
/// the Exception to the given Exception Handler `$handler` with its arguments
/// `$handler_args`. The `$handler` function must be defined in the same scope
/// as `sass-errors-try-catch`.
///
/// If an Exception has been thrown, returns `false. Otherwise, returns `$value`.
///
/// @group API
/// @see sass-errors-try
/// @see sass-errors-catch
///
/// @param" data-type="css"><a href="#undefined-css-/// If an Exception has been thrown, catch it with `sass-errors-catch` and pass
/// the Exception to the given Exception Handler `$handler` with its arguments
/// `$handler_args`. The `$handler` function must be defined in the same scope
/// as `sass-errors-try-catch`.
///
/// If an Exception has been thrown, returns `false. Otherwise, returns `$value`.
///
/// @group API
/// @see sass-errors-try
/// @see sass-errors-catch
///
/// @param">/// If an Exception has been thrown, catch it with `sass-errors-catch` and pass /// the Exception to the given Exception Handler `$handler` with its arguments /// `$handler_args`. The `$handler` function must be defined in the same scope /// as `sass-errors-try-catch`. /// /// If an Exception has been thrown, returns `false. Otherwise, returns `$value`. /// /// @group API /// @see sass-errors-try /// @see sass-errors-catch /// /// @param</a></li></ul></div></div></aside><article class="main" role="main"><header class="header" role="banner"><div class="container"><div class="sassdoc__searchbar searchbar"><label for="js-search-input" class="visually-hidden">Search</label><div class="searchbar__form" id="js-search"><input name="search" type="search" class="searchbar__field" autocomplete="off" autofocus id="js-search-input" placeholder="Search"><ul class="searchbar__suggestions" id="js-search-suggestions"></ul></div></div></div></header><section class="main__section"><h1 class="main__heading" id="alias"><div class="container">Aliases</div></h1><section class="main__sub-section" id="alias-function"><h2 class="main__heading--secondary"><div class="container">functions</div></h2><section class="main__item container item item--alias" id="alias-function-se-init"><h3 class="item__heading"><a class="item__name" href="#function-se-init">se-init</a> <span class="item__alias">(alias for <a href="#function-sass-errors-init"><code>sass-errors-init</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-init() { 
  @return sass-errors-init($args...);
 }" data-collapsed="@function se-init() { ... }"><code>@function se-init() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-init">sass-errors-init</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-store"><h3 class="item__heading"><a class="item__name" href="#function-se-store">se-store</a> <span class="item__alias">(alias for <a href="#function-sass-errors-store"><code>sass-errors-store</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-store() { 
  @return sass-errors-store($args...);
 }" data-collapsed="@function se-store() { ... }"><code>@function se-store() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-store">sass-errors-store</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-message"><h3 class="item__heading"><a class="item__name" href="#function-se-message">se-message</a> <span class="item__alias">(alias for <a href="#function-sass-errors-message"><code>sass-errors-message</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-message() { 
  @return sass-errors-message($args...);
 }" data-collapsed="@function se-message() { ... }"><code>@function se-message() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-message">sass-errors-message</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-new"><h3 class="item__heading"><a class="item__name" href="#function-se-new">se-new</a> <span class="item__alias">(alias for <a href="#function-sass-errors-new"><code>sass-errors-new</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-new() { 
  @return sass-errors-new($args...);
 }" data-collapsed="@function se-new() { ... }"><code>@function se-new() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-new">sass-errors-new</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-throw"><h3 class="item__heading"><a class="item__name" href="#function-se-throw">se-throw</a> <span class="item__alias">(alias for <a href="#function-sass-errors-throw"><code>sass-errors-throw</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-throw() { 
  @return sass-errors-throw($args...);
 }" data-collapsed="@function se-throw() { ... }"><code>@function se-throw() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-throw">sass-errors-throw</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-try"><h3 class="item__heading"><a class="item__name" href="#function-se-try">se-try</a> <span class="item__alias">(alias for <a href="#function-sass-errors-try"><code>sass-errors-try</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-try() { 
  @return sass-errors-try($args...);
 }" data-collapsed="@function se-try() { ... }"><code>@function se-try() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-try">sass-errors-try</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-catch"><h3 class="item__heading"><a class="item__name" href="#function-se-catch">se-catch</a> <span class="item__alias">(alias for <a href="#function-sass-errors-catch"><code>sass-errors-catch</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-catch() { 
  @return sass-errors-catch($args...);
 }" data-collapsed="@function se-catch() { ... }"><code>@function se-catch() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-catch">sass-errors-catch</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-try-catch"><h3 class="item__heading"><a class="item__name" href="#function-se-try-catch">se-try-catch</a> <span class="item__alias">(alias for <a href="#function-sass-errors-try-catch"><code>sass-errors-try-catch</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-try-catch() { 
  @return sass-errors-try-catch($args...);
 }" data-collapsed="@function se-try-catch() { ... }"><code>@function se-try-catch() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-try-catch">sass-errors-try-catch</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-get"><h3 class="item__heading"><a class="item__name" href="#function-se-get">se-get</a> <span class="item__alias">(alias for <a href="#function-sass-errors-get"><code>sass-errors-get</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-get() { 
  @return sass-errors-get($args...);
 }" data-collapsed="@function se-get() { ... }"><code>@function se-get() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-get">sass-errors-get</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-is-exception"><h3 class="item__heading"><a class="item__name" href="#function-se-is-exception">se-is-exception</a> <span class="item__alias">(alias for <a href="#function-sass-errors-is-exception"><code>sass-errors-is-exception</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-is-exception() { 
  @return sass-errors-is-exception($args...);
 }" data-collapsed="@function se-is-exception() { ... }"><code>@function se-is-exception() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-is-exception">sass-errors-is-exception</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-debug"><h3 class="item__heading"><a class="item__name" href="#function-se-debug">se-debug</a> <span class="item__alias">(alias for <a href="#function-sass-errors-debug"><code>sass-errors-debug</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-debug() { 
  @return sass-errors-debug($args...);
 }" data-collapsed="@function se-debug() { ... }"><code>@function se-debug() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-debug">sass-errors-debug</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-error"><h3 class="item__heading"><a class="item__name" href="#function-se-error">se-error</a> <span class="item__alias">(alias for <a href="#function-sass-errors-error"><code>sass-errors-error</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-error() { 
  @return sass-errors-error($args...);
 }" data-collapsed="@function se-error() { ... }"><code>@function se-error() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-error">sass-errors-error</a>.</p></section><section class="main__item container item item--alias" id="alias-function-se-warn"><h3 class="item__heading"><a class="item__name" href="#function-se-warn">se-warn</a> <span class="item__alias">(alias for <a href="#function-sass-errors-warn"><code>sass-errors-warn</code></a>)</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function se-warn() { 
  @return sass-errors-warn($args...);
 }" data-collapsed="@function se-warn() { ... }"><code>@function se-warn() { ... }</code></pre></div><p>Refer to <a href="#function-sass-errors-warn">sass-errors-warn</a>.</p></section></section></section><section class="main__section"><h1 class="main__heading" id="api"><div class="container">API</div></h1><section class="main__sub-section" id="api-function"><h2 class="main__heading--secondary"><div class="container">functions</div></h2><section class="main__item container item" id="api-function-sass-errors-catch"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-catch">sass-errors-catch</a> <span class="item__aliased">(aliased as <a href="#function-se-catch"><code>se-catch</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-catch($types...) { 
  $e: $-sass-errors-exception !global;

  @if $e != null {

    $type: map-get($e, &#39;type&#39;);
    $is-types: length($types) != 0;

    @if not $is-types or ($is-types and index($types, $type) != null) {
      $-sass-errors-exception: null !global;
      @return $e;
    }

  }
  @return null;
 }" data-collapsed="@function sass-errors-catch($types...) { ... }"><code>@function sass-errors-catch($types...) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>If an Exception has been throw, catch it and return it. The Exception will not be catched again.</p><p>If types <code>$types</code> are given, return the Exception only if its type matches with one of the given types. Otherwise, return <code>null</code>.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$types...</code></th><td data-label="desc"><p>Type or types the Exception type must match to.</p></td><td data-label="type"><code>String</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr></tbody></table><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>Exception</code> or <code>Null</code> &mdash;<p>Catched Exception.</p></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><div class="example__description"><p>Catch an Exception</p></div><pre class="example__code language-scss"><code>@function my-function-with-exception() {
  @return sass-errors-throw(&#39;Error&#39;, &#39;my_error_type&#39;);
}

$ret: my-function-with-exception();
$e: sass-errors-catch();
// &#39;my_error_type&#39; Exception</code></pre></div><div class="item__example example"><div class="example__description"><p>Try to catch several times the same Exception</p></div><pre class="example__code language-scss"><code>$ret: my-function-with-exception();
$e: sass-errors-catch();
// &#39;my_error_type&#39; Exception
$e: sass-errors-catch();
// null</code></pre></div><div class="item__example example"><div class="example__description"><p>Catch specific types of Exceptions (with a type matching)</p></div><pre class="example__code language-scss"><code>$ret: my-function-with-exception();
$e: sass-errors-catch(&#39;a_type&#39;, &#39;my_error_type&#39;);
// &#39;my_error_type&#39; Exception</code></pre></div><div class="item__example example"><div class="example__description"><p>Catch specific types of Exceptions (with no types matching)</p></div><pre class="example__code language-scss"><code>$ret: my-function-with-exception();
$e: sass-errors-catch(&#39;a_type&#39;, &#39;an_other_type&#39;);
// null</code></pre></div><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-catch"><code>se-catch</code></a></li><li><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-try-catch"><code>sass-errors-try-catch</code></a></li></ul><h3 class="item__sub-heading">See</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#function-sass-errors-throw"><code>sass-errors-throw</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-get"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-get">sass-errors-get</a> <span class="item__aliased">(aliased as <a href="#function-se-get"><code>se-get</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-get() { 
  @return $-sass-errors-exception;
 }" data-collapsed="@function sass-errors-get() { ... }"><code>@function sass-errors-get() { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Returns current thrown and not catched Exception. If not Exception has been thrown or is not already catchen, returns <code>null</code></p></div><h3 class="item__sub-heading">Parameters</h3><p>None.</p><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>Exception</code> or <code>Null</code> &mdash;<p>Current thrown and not catched Exception.</p></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><div class="example__description"><p>Get an Exception</p></div><pre class="example__code language-scss"><code>@function my-function-with-exception() {
  @return sass-errors-throw(&#39;Error&#39;, &#39;my_error_type&#39;);
}

$ret: my-function-with-exception();
$e: sass-errors-get();
// &#39;my_error_type&#39; Exception</code></pre></div><div class="item__example example"><div class="example__description"><p>Get several times the same Exception</p></div><pre class="example__code language-scss"><code>$ret: my-function-with-exception();
$e: sass-errors-get();
// &#39;my_error_type&#39; Exception
$e: sass-errors-get();
// &#39;my_error_type&#39; Exception</code></pre></div><div class="item__example example"><div class="example__description"><p>Get an Exception after it has been catched</p></div><pre class="example__code language-scss"><code>$ret: my-function-with-exception();
$e: sass-errors-catch();
// &#39;my_error_type&#39; Exception
$e: sass-errors-get();
// null</code></pre></div><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-get"><code>se-get</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-init"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-init">sass-errors-init</a> <span class="item__aliased">(aliased as <a href="#function-se-init"><code>se-init</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-init($prefix: null) { 
  $-sass-errors-exception: null !global;
  $-sass-errors-prefix: $prefix !global;

  @return null;
 }" data-collapsed="@function sass-errors-init($prefix: null) { ... }"><code>@function sass-errors-init($prefix: null) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Initialize Sass Errors required variables and options</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$prefix</code></th><td data-label="desc"><p>Default prefix for new Exceptions</p></td><td data-label="type"><code>String</code></td><td data-label="default"><code>null</code></td></tr></tbody></table><h3 class="item__sub-heading">Example</h3><div class="item__example example"><pre class="example__code language-scss"><code>$_: sass-errors-init();</code></pre></div><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-init"><code>se-init</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-is-exception"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-is-exception">sass-errors-is-exception</a> <span class="item__aliased">(aliased as <a href="#function-se-is-exception"><code>se-is-exception</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-is-exception() { 
  @return ($-sass-errors-exception != null);
 }" data-collapsed="@function sass-errors-is-exception() { ... }"><code>@function sass-errors-is-exception() { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Returns <code>true</code> if an Exception has been thrown and not catched, <code>false</code> otherwise.</p></div><h3 class="item__sub-heading">Parameters</h3><p>None.</p><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>Boolean</code></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><div class="example__description"><p>Check if an Exception exists</p></div><pre class="example__code language-scss"><code>@function my-function-with-exception() {
  @return sass-errors-throw(&#39;Error&#39;, &#39;my_error_type&#39;);
}

$is_e: sass-errors-is-exception();
// `false`

$ret: my-function-with-exception();
$is_e: sass-errors-is-exception();
// `true`

$e: sass-errors-catch();
$is_e: sass-errors-is-exception();
// `false`</code></pre></div><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-is-exception"><code>se-is-exception</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-message"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-message">sass-errors-message</a> <span class="item__aliased">(aliased as <a href="#function-se-message"><code>se-message</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-message($error, $prefix: null) { 
  @if type-of($prefix) == &#39;string&#39; and str-length($prefix) &gt; 0 {
    @return &#39;#{$prefix}: #{$error}&#39;;
  }
  @else {
    @return &#39;#{$error}&#39;;
  }
 }" data-collapsed="@function sass-errors-message($error, $prefix: null) { ... }"><code>@function sass-errors-message($error, $prefix: null) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Makes a formatted message from an error message and a prefix and returns it.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$error</code></th><td data-label="desc"><p>Error message</p></td><td data-label="type"><code>String</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$prefix</code></th><td data-label="desc"><p>Message prefix</p></td><td data-label="type"><code>String</code></td><td data-label="default"><code>null</code></td></tr></tbody></table><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>String</code></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><div class="example__description"><p>Simple message</p></div><pre class="example__code language-scss"><code>$msg: sass-errors-message(&#39;Error&#39;);
// &#39;Error&#39;</code></pre></div><div class="item__example example"><div class="example__description"><p>Message with prefix</p></div><pre class="example__code language-scss"><code>$msg: sass-errors-message(&#39;Error&#39;, &#39;my-function&#39;);
// &#39;my-function: Error&#39;</code></pre></div><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-message"><code>se-message</code></a></li><li><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-new"><code>sass-errors-new</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-new"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-new">sass-errors-new</a> <span class="item__aliased">(aliased as <a href="#function-se-new"><code>se-new</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-new($error, $type, $datas: (), $prefix: $-sass-errors-prefix) { 
  @if type-of($error) == &#39;map&#39; {
    @return $error;
  }

  $message: sass-errors-message($error, $prefix);

  @return (
    message: $message,
    error: $error,
    type: $type,
    datas: $datas
  );
 }" data-collapsed="@function sass-errors-new($error, $type, $datas: (), $prefix: $-sass-errors-prefix) { ... }"><code>@function sass-errors-new($error, $type, $datas: (), $prefix: $-sass-errors-prefix) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Creates an Exception object and returns it. If an Exception is passed as first argument, returns it.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$error</code></th><td data-label="desc"><p>Error message</p></td><td data-label="type"><code>String</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$type</code></th><td data-label="desc"><p>Exception type</p></td><td data-label="type"><code>String</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$datas</code></th><td data-label="desc"><p>Map of relevant datas to be embedded in the Exception</p></td><td data-label="type"><code>Map</code></td><td data-label="default"><code>()</code></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$prefix</code></th><td data-label="desc"><p>Message prefix, global prefix by default (see <code>sass-errors-init</code>)</p></td><td data-label="type"><code>String</code></td><td data-label="default"><code>$-sass-errors-prefix</code></td></tr></tbody></table><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>Exception</code></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><pre class="example__code language-scss"><code>@function my-function($value) {
  $e: sass-errors-new(&#39;Error&#39;, &#39;unknown_error&#39;, (value: $value), &#39;my-function&#39;);

  // (
  //   message: &#39;my-function: Error&#39;,
  //   error: &#39;Error&#39;,
  //   type: &#39;unknown_error&#39;,
  //   datas: (value: ...),
  // )
}</code></pre></div><h3 class="item__sub-heading">Requires</h3><ul class="list-unstyled"><li class="item__description item__description--inline"><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-message"><code>sass-errors-message</code></a></li></ul><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-new"><code>se-new</code></a></li><li><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-store"><code>sass-errors-store</code></a></li><li><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-throw"><code>sass-errors-throw</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-store"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-store">sass-errors-store</a> <span class="item__aliased">(aliased as <a href="#function-se-store"><code>se-store</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-store($obj, $args...: ()...) { 
  $e: sass-errors-new($args...);

  @if type-of($obj) != &#39;list&#39; {
    $obj: ();
  }
  $obj: append($obj, $e);

  @return $obj;
 }" data-collapsed="@function sass-errors-store($obj, $args...: ()...) { ... }"><code>@function sass-errors-store($obj, $args...: ()...) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Create an Exception, append it to the given store <code>$obj</code> and return it. The new exception is not thrown. If an Exception is passed as second argument, store it.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$obj</code></th><td data-label="desc"><p>List where to store the new Exception</p></td><td data-label="type"><code>List</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$args...</code></th><td data-label="desc"><p>Arguments used to create an Exception. See <code>@function sass-errors-new</code>.</p></td><td data-label="type"><code>Arglist</code></td><td data-label="default"><code>()</code></td></tr></tbody></table><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>List</code> &mdash;<p>The new list</p></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><pre class="example__code language-scss"><code>$store: ();
$store: sass-errors-new($store, &#39;Error 1&#39;, &#39;unknown_error&#39;);
$store: sass-errors-new($store, &#39;Error 2&#39;, &#39;unknown_error&#39;);

// (
//   (
//     message: &#39;Error 1&#39;,
//     error: &#39;Error 1,
//     type: &#39;unknown_error&#39;,
//     datas: ()
//   ),
//   (
//     message: &#39;Error 2&#39;,
//     error: &#39;Error 1,
//     type: &#39;unknown_error&#39;,
//     datas: ()
//   )
// )</code></pre></div><h3 class="item__sub-heading">Requires</h3><ul class="list-unstyled"><li class="item__description item__description--inline"><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-new"><code>sass-errors-new</code></a></li></ul><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-store"><code>se-store</code></a></li></ul><h3 class="item__sub-heading">See</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#function-sass-errors-new"><code>sass-errors-new</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-throw"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-throw">sass-errors-throw</a> <span class="item__aliased">(aliased as <a href="#function-se-throw"><code>se-throw</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-throw($args...: ()...) { 
  $e: sass-errors-new($args...);

  $-sass-errors-exception: $e !global;

  @return false;
 }" data-collapsed="@function sass-errors-throw($args...: ()...) { ... }"><code>@function sass-errors-throw($args...: ()...) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Create an Exception and throw it. If an Exception is passed as first parameter, throw it. Always return false: this function should be called with <code>@return</code>.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$args...</code></th><td data-label="desc"><p>Arguments used to create an Exception. See <code>@function sass-errors-new</code></p></td><td data-label="type"><code>Arglist</code></td><td data-label="default"><code>()</code></td></tr></tbody></table><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>False</code></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><pre class="example__code language-scss"><code>@function my-function() {
  @return sass-errors-throw(&#39;Error&#39;, &#39;unknown_error&#39;);
}</code></pre></div><h3 class="item__sub-heading">Requires</h3><ul class="list-unstyled"><li class="item__description item__description--inline"><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-new"><code>sass-errors-new</code></a></li></ul><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-throw"><code>se-throw</code></a></li><li><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-try"><code>sass-errors-try</code></a></li></ul><h3 class="item__sub-heading">See</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#function-sass-errors-new"><code>sass-errors-new</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-try-catch"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-try-catch">sass-errors-try-catch</a> <span class="item__aliased">(aliased as <a href="#function-se-try-catch"><code>se-try-catch</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-try-catch($value, $catch_func_name: sass-errors-error, $catch_func_args: ()..., $args...: ()...) { 
  $res: sass-errors-try($value, $args...);
  $e: sass-errors-catch($types...);

  @if $e and $handler {
    $_: call($handler, $e, $handler_args...);
  }
  @return $res;
 }" data-collapsed="@function sass-errors-try-catch($value, $catch_func_name: sass-errors-error, $catch_func_args: ()..., $args...: ()...) { ... }"><code>@function sass-errors-try-catch($value, $catch_func_name: sass-errors-error, $catch_func_args: ()..., $args...: ()...) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>If an Exception has been thrown, catch it with <code>sass-errors-catch</code> and pass the Exception to the given Exception Handler <code>$handler</code> with its arguments <code>$handler_args</code>. The <code>$handler</code> function must be defined in the same scope as <code>sass-errors-try-catch</code>.</p><p>If an Exception has been thrown, returns <code>false. Otherwise, returns</code>$value`.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$value</code></th><td data-label="desc"><p>Value to check.</p></td><td data-label="type"><code>Any</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$catch_func_name</code></th><td data-label="desc"><p>Name of the function to which the Exception is passed.</p></td><td data-label="type"><code>String</code></td><td data-label="default"><code>sass-errors-error</code></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$catch_func_args</code></th><td data-label="desc"><p>Arguments of the catch function, following the Exception.</p></td><td data-label="type"><code>Arglist</code></td><td data-label="default"><code>()</code></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$args...</code></th><td data-label="desc"><p>Arguments used to create an Exception. See <code>@function sass-errors-new</code>.</p></td><td data-label="type"><code>Arglist</code></td><td data-label="default"><code>()</code></td></tr></tbody></table><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>*</code> or <code>False</code> &mdash;<p>The function return.</p></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><pre class="example__code language-scss"><code>@function my-try-function-with-value() {
  @return &#39;simple_value&#39;;
}

@function my-try-function-with-exception() {
  @return sass-errors-throw(&#39;Error&#39;, &#39;my_error_type&#39;);
}

@function my-catch-function($e) {
  @debug $e;
}

$ret: sass-errors-try-catch(my-try-function-with-value(), my-catch-function, ());
// &#39;simple value&#39;

$ret: sass-errors-try-catch(my-try-function-with-exception(), my-catch-function, ());
// $ret: `false`
// @debug: &#39;my_error_type&#39; Exception</code></pre></div><h3 class="item__sub-heading">Requires</h3><ul class="list-unstyled"><li class="item__description item__description--inline"><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-try"><code>sass-errors-try</code></a></li><li class="item__description item__description--inline"><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-catch"><code>sass-errors-catch</code></a></li></ul><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-try-catch"><code>se-try-catch</code></a></li></ul><h3 class="item__sub-heading">See</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#function-sass-errors-try"><code>sass-errors-try</code></a></li><li><span class="item__cross-type">[function]</span> <a href="#function-sass-errors-catch"><code>sass-errors-catch</code></a></li></ul></section><section class="main__item container item" id="api-function-sass-errors-try"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-try">sass-errors-try</a> <span class="item__aliased">(aliased as <a href="#function-se-try"><code>se-try</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-try($value, $args...: ()...) { 

  $e: $-sass-errors-exception;
  @if $e != null {
    @return false;
  }

  @if $value == false {
    @if length($args) == 0 and length(keywords($args)) == 0 {
      $args: (
        error: &#39;Unknown error. Expected given $value to not be false.&#39;,
        type: &#39;unknown_error&#39;
      );
    }
    @return sass-errors-throw($args...);
  }

  @return $value;
 }" data-collapsed="@function sass-errors-try($value, $args...: ()...) { ... }"><code>@function sass-errors-try($value, $args...: ()...) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>If the passed value <code>$value</code> is equal to <code>false</code> and no exception has been thrown, throw a new <code>unknown_error</code> Exception. If an Exception has been thrown, returns <code>false. Otherwise, returns</code>$value.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$value</code></th><td data-label="desc"><p>Value to check.</p></td><td data-label="type"><code>Any</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr><tr class="item__parameter"><th scope="row" data-label="name"><code>$args...</code></th><td data-label="desc"><p>Arguments used to create an Exception. See <code>@function sass-errors-new</code>.</p></td><td data-label="type"><code>Arglist</code></td><td data-label="default"><code>()</code></td></tr></tbody></table><h3 class="item__sub-heading">Returns</h3><div class="item__description item__description--inline"><code>*</code> or <code>False</code> &mdash;<p>The given value, or <code>false</code> if an exception has been thrown.</p></div><h3 class="item__sub-heading">Example</h3><div class="item__example example"><div class="example__description"><p>A non-false value is returned</p></div><pre class="example__code language-scss"><code>@function my-function-with-value() {
  @return &#39;simple_value&#39;;
}

$ret: sass-errors-try(my-function-with-value());
// &#39;simple value&#39;
$e: sass-errors-catch();
// null</code></pre></div><div class="item__example example"><div class="example__description"><p><code>false</code> is returned</p></div><pre class="example__code language-scss"><code>@function my-function-with-false() {
  @return false;
}

$ret: sass-errors-try(my-function-with-false());
// false
$e: sass-errors-catch();
// &#39;unknown_error&#39; Exception</code></pre></div><div class="item__example example"><div class="example__description"><p>Setting the Exception to be throw when <code>false</code> is returned</p></div><pre class="example__code language-scss"><code>@function my-function-with-false() {
  @return false;
}

$ret: sass-errors-try(
  my-function-with-false(),
  &#39;My function failed&#39;,
  &#39;my_function_failed&#39;
);
// false
$e: sass-errors-catch();
// &#39;my_function_failed&#39; Exception</code></pre></div><div class="item__example example"><div class="example__description"><p>An Exception is thrown</p></div><pre class="example__code language-scss"><code>@function my-function-with-exception() {
  @return sass-errors-throw(&#39;Error&#39;, &#39;my_error_type&#39;);
}

$ret: sass-errors-try(my-function-with-exception());
// false
$e: sass-errors-catch();
// &#39;my_error_type&#39; Exception</code></pre></div><h3 class="item__sub-heading">Requires</h3><ul class="list-unstyled"><li class="item__description item__description--inline"><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-throw"><code>sass-errors-throw</code></a></li></ul><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-try"><code>se-try</code></a></li><li><span class="item__cross-type">[function]</span> <a href="#api-function-sass-errors-try-catch"><code>sass-errors-try-catch</code></a></li></ul></section></section></section><section class="main__section"><h1 class="main__heading" id="handler"><div class="container">Exception Handlers</div></h1><section class="main__sub-section" id="handler-function"><h2 class="main__heading--secondary"><div class="container">functions</div></h2><section class="main__item container item" id="handler-function-sass-errors-debug"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-debug">sass-errors-debug</a> <span class="item__aliased">(aliased as <a href="#function-se-debug"><code>se-debug</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-debug($e) { 
  $msg: -sass-errors-get-exception-message($e);
  @if ($msg) {
    @debug $msg;
  }

  @return null;
 }" data-collapsed="@function sass-errors-debug($e) { ... }"><code>@function sass-errors-debug($e) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Display the message of the given Exception <code>$e</code> with @debug. This function is an Exception Handler and can be passed as a catch function to <code>sass-errors-try-catch</code>.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$e</code></th><td data-label="desc">&mdash;<span class="visually-hidden">none</span></td><td data-label="type"><code>Exception</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr></tbody></table><h3 class="item__sub-heading">Example</h3><div class="item__example example"><pre class="example__code language-scss"><code>$e: sass-errors-throw(&#39;Error&#39;, $prefix: &#39;my-function&#39;);
$_: sass-errors-debug($e);
// @debug: &#39;my-function: Error&#39;</code></pre></div><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-debug"><code>se-debug</code></a></li></ul><h3 class="item__sub-heading">See</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#function-sass-errors-try-catch"><code>sass-errors-try-catch</code></a></li></ul></section><section class="main__item container item" id="handler-function-sass-errors-error"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-error">sass-errors-error</a> <span class="item__aliased">(aliased as <a href="#function-se-error"><code>se-error</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-error($e) { 
  $msg: -sass-errors-get-exception-message($e);
  @if ($msg) {
    @error $msg;
  }

  @return null;
 }" data-collapsed="@function sass-errors-error($e) { ... }"><code>@function sass-errors-error($e) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Display the message of the given Exception <code>$e</code> with @error. This function is an Exception Handler and can be passed as a catch function to <code>sass-errors-try-catch</code>.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$e</code></th><td data-label="desc">&mdash;<span class="visually-hidden">none</span></td><td data-label="type"><code>Exception</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr></tbody></table><h3 class="item__sub-heading">Example</h3><div class="item__example example"><pre class="example__code language-scss"><code>$e: sass-errors-throw(&#39;Error&#39;, $prefix: &#39;my-function&#39;);
$_: sass-errors-error($e);
// @error: &#39;my-function: Error&#39;</code></pre></div><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-error"><code>se-error</code></a></li></ul><h3 class="item__sub-heading">See</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#function-sass-errors-try-catch"><code>sass-errors-try-catch</code></a></li></ul></section><section class="main__item container item" id="handler-function-sass-errors-warn"><h3 class="item__heading"><a class="item__name" href="#function-sass-errors-warn">sass-errors-warn</a> <span class="item__aliased">(aliased as <a href="#function-se-warn"><code>se-warn</code></a> )</span></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@function sass-errors-warn($e) { 
  $msg: -sass-errors-get-exception-message($e);
  @if ($msg) {
    @warn $msg;
  }

  @return null;
 }" data-collapsed="@function sass-errors-warn($e) { ... }"><code>@function sass-errors-warn($e) { ... }</code></pre></div><h3 class="item__sub-heading">Description</h3><div class="item__description"><p>Display the message of the given Exception <code>$e</code> with @warn. This function is an Exception Handler and can be passed as a catch function to <code>sass-errors-try-catch</code>.</p></div><h3 class="item__sub-heading">Parameters</h3><table class="item__parameters"><thead><tr><th scope="col"><span class="visually-hidden">parameter</span>Name</th><th scope="col"><span class="visually-hidden">parameter</span>Description</th><th scope="col"><span class="visually-hidden">parameter</span>Type</th><th scope="col"><span class="visually-hidden">parameter</span>Default value</th></tr></thead><tbody><tr class="item__parameter"><th scope="row" data-label="name"><code>$e</code></th><td data-label="desc">&mdash;<span class="visually-hidden">none</span></td><td data-label="type"><code>Exception</code></td><td data-label="default">&mdash;<span class="visually-hidden">none</span></td></tr></tbody></table><h3 class="item__sub-heading">Example</h3><div class="item__example example"><pre class="example__code language-scss"><code>$e: sass-errors-throw(&#39;Error&#39;, $prefix: &#39;my-function&#39;);
$_: sass-errors-debug($e);
// @warn: &#39;my-function: Error&#39;</code></pre></div><h3 class="item__sub-heading">Used by</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#alias-function-se-warn"><code>se-warn</code></a></li></ul><h3 class="item__sub-heading">See</h3><ul class="list-unstyled"><li><span class="item__cross-type">[function]</span> <a href="#function-sass-errors-try-catch"><code>sass-errors-try-catch</code></a></li></ul></section></section></section><section class="main__section"><h1 class="main__heading" id="undefined"><div class="container">General</div></h1><section class="main__sub-section" id="undefined-css"><h2 class="main__heading--secondary"><div class="container">csss</div></h2><section class="main__item container item" id="undefined-css-/// If an Exception has been thrown, catch it with `sass-errors-catch` and pass
/// the Exception to the given Exception Handler `$handler` with its arguments
/// `$handler_args`. The `$handler` function must be defined in the same scope
/// as `sass-errors-try-catch`.
///
/// If an Exception has been thrown, returns `false. Otherwise, returns `$value`.
///
/// @group API
/// @see sass-errors-try
/// @see sass-errors-catch
///
/// @param"><h3 class="item__heading"><a class="item__name" href="#css-/// If an Exception has been thrown, catch it with `sass-errors-catch` and pass
/// the Exception to the given Exception Handler `$handler` with its arguments
/// `$handler_args`. The `$handler` function must be defined in the same scope
/// as `sass-errors-try-catch`.
///
/// If an Exception has been thrown, returns `false. Otherwise, returns `$value`.
///
/// @group API
/// @see sass-errors-try
/// @see sass-errors-catch
///
/// @param">/// If an Exception has been thrown, catch it with `sass-errors-catch` and pass /// the Exception to the given Exception Handler `$handler` with its arguments /// `$handler_args`. The `$handler` function must be defined in the same scope /// as `sass-errors-try-catch`. /// /// If an Exception has been thrown, returns `false. Otherwise, returns `$value`. /// /// @group API /// @see sass-errors-try /// @see sass-errors-catch /// /// @param</a></h3><div class="item__code-wrapper"><pre class="item__code item__code--togglable language-scss" data-current-state="collapsed" data-expanded="@css /// If an Exception has been thrown, catch it with `sass-errors-catch` and pass
/// the Exception to the given Exception Handler `$handler` with its arguments
/// `$handler_args`. The `$handler` function must be defined in the same scope
/// as `sass-errors-try-catch`.
///
/// If an Exception has been thrown, returns `false. Otherwise, returns `$value`.
///
/// @group API
/// @see sass-errors-try
/// @see sass-errors-catch
///
/// @param() {  }" data-collapsed="@css /// If an Exception has been thrown, catch it with `sass-errors-catch` and pass
/// the Exception to the given Exception Handler `$handler` with its arguments
/// `$handler_args`. The `$handler` function must be defined in the same scope
/// as `sass-errors-try-catch`.
///
/// If an Exception has been thrown, returns `false. Otherwise, returns `$value`.
///
/// @group API
/// @see sass-errors-try
/// @see sass-errors-catch
///
/// @param() { ... }"><code>@css /// If an Exception has been thrown, catch it with `sass-errors-catch` and pass
/// the Exception to the given Exception Handler `$handler` with its arguments
/// `$handler_args`. The `$handler` function must be defined in the same scope
/// as `sass-errors-try-catch`.
///
/// If an Exception has been thrown, returns `false. Otherwise, returns `$value`.
///
/// @group API
/// @see sass-errors-try
/// @see sass-errors-catch
///
/// @param() { ... }</code></pre></div></section></section></section><footer class="footer" role="contentinfo"><div class="container"><div class="footer__project-info project-info"><!-- Name and URL --><a class="project-info__name" href="https://github.com/ncoden/sass-errors#readme">Sass Errors</a><!-- Version --> <span class="project-info__version">- v0.4.0</span><!-- License --> <span class="project-info__license">, under MIT</span></div><a class="footer__watermark" href="http://sassdoc.com"><img src="assets/images/logo_light_inline.svg" alt="SassDoc Logo"></a></div></footer></article><script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script><script>window.jQuery || document.write('<script src="assets/js/vendor/jquery.min.js"><\/script>')</script><script src="assets/js/main.min.js"></script></body></html>